internal/source/image.go

```diff
@@ -0,0 +1,188 @@
+			"core.rukpak.io/owner-name": bundle.Name,
+		})
+		pod.SetOwnerReferences([]metav1.OwnerReference{*controllerRef})
+		pod.Spec.AutomountServiceAccountToken = &automountServiceAccountToken
+		pod.Spec.Volumes = []corev1.Volume{
+			{Name: "util", VolumeSource: corev1.VolumeSource{EmptyDir: &corev1.EmptyDirVolumeSource{}}},
+		}
+		pod.Spec.RestartPolicy = corev1.RestartPolicyNever
+		if len(pod.Spec.InitContainers) != 1 {
+			pod.Spec.InitContainers = make([]corev1.Container, 1)
+		}
+		pod.Spec.InitContainers[0].Name = "install-unpacker"
+		pod.Spec.InitContainers[0].Image = i.UnpackImage
+		pod.Spec.InitContainers[0].ImagePullPolicy = corev1.PullIfNotPresent
+		pod.Spec.InitContainers[0].Command = []string{"cp", "-Rv", "/unpack", "/util/unpack"}
+		pod.Spec.InitContainers[0].VolumeMounts = []corev1.VolumeMount{{Name: "util", MountPath: "/util"}}
+
+		if len(pod.Spec.Containers) != 1 {
+			pod.Spec.Containers = make([]corev1.Container, 1)
+		}
+		pod.Spec.Containers[0].Name = bundleUnpackContainerName
+		pod.Spec.Containers[0].Image = bundle.Spec.Source.Image.Ref
+		pod.Spec.Containers[0].Command = []string{"/bin/unpack", "--bundle-dir", "/"}
+		pod.Spec.Containers[0].VolumeMounts = []corev1.VolumeMount{{Name: "util", MountPath: "/bin"}}
+		return nil
+	})
+}
+
+func pendingPodResult(pod *corev1.Pod) *Result {
+	var messages []string
+	for _, cStatus := range append(pod.Status.InitContainerStatuses, pod.Status.ContainerStatuses...) {
+		if cStatus.State.Waiting != nil && cStatus.State.Waiting.Reason == "ErrImagePull" {
+			messages = append(messages, cStatus.State.Waiting.Message)
+		}
+		if cStatus.State.Waiting != nil && cStatus.State.Waiting.Reason == "ImagePullBackOff" {
+			messages = append(messages, cStatus.State.Waiting.Message)
+		}
+	}
+	return &Result{State: StatePending, Message: strings.Join(messages, "; ")}
+}
+
+func (i *Image) failedPodResult(ctx context.Context, pod *corev1.Pod) error {
+	logs, err := i.getPodLogs(ctx, pod)
+	if err != nil {
+		return fmt.Errorf("unpack failed: failed to retrieve failed pod logs: %w", err)
+	}
+	_ = i.Client.Delete(ctx, pod)
+	return fmt.Errorf("unpack failed: %v", string(logs))
+}
+
+func (i *Image) succeededPodResult(ctx context.Context, pod *corev1.Pod) (*Result, error) {
+	bundleFS, err := i.getBundleContents(ctx, pod)
+	if err != nil {
+		return nil, fmt.Errorf("get bundle contents: %w", err)
+	}
+
+	digest, err := i.getBundleImageDigest(pod)
+	if err != nil {
+		return nil, fmt.Errorf("get bundle image digest: %w", err)
+	}
+
+	resolvedSource := &rukpakv1alpha1.BundleSource{
+		Type:  rukpakv1alpha1.BundleSourceTypeImage,
+		Image: &rukpakv1alpha1.ImageSource{Ref: digest},
+	}
+
+	return &Result{Bundle: bundleFS, ResolvedSource: resolvedSource, State: StateUnpacked}, nil
+}
+
+func (i *Image) getBundleContents(ctx context.Context, pod *corev1.Pod) (fs.FS, error) {
+	bundleData, err := i.getPodLogs(ctx, pod)
+	if err != nil {
+		return nil, fmt.Errorf("get bundle contents: %w", err)
+	}
```

Comment:
nit: Pass the `bundleData []byte` as a parameter instead?

Reply:
I'm just going to mark this as resolved as this is a nit and an issue with the existing implementation.